File: ports/csharp/src/core/thread/BcsThread.cs
using System.Threading;
using System.Threading.Tasks;

namespace BCS.Core.Thread {
    // BcsThread abstracts thread execution, mapped to C# Tasks and Threads
    public class BcsThread {
        private Task _task;

        public void Start(System.Action action) {
            _task = Task.Run(action);
        }

        public void Wait() {
            if (_task != null) {
                _task.Wait();
            }
        }

        public static void Sleep(int ms) {
            System.Threading.Thread.Sleep(ms);
        }
    }
}

File: ports/java/src/core/thread/BcsThread.java
package bcs.core.thread;

// BcsThread abstracts standard thread execution for Java
public class BcsThread {
    private Thread thread;

    public void start(Runnable runnable) {
        this.thread = new Thread(runnable);
        this.thread.start();
    }

    public void waitThread() {
        if (this.thread != null) {
            try {
                this.thread.join();
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
            }
        }
    }

    public static void sleep(int ms) {
        try {
            Thread.sleep(ms);
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
    }
}
